apps/desktop/src/app/upload/settings/settings.component.html

```diff
@@ -137,10 +137,24 @@ <h2 class="unit-test">{{ 'upload.source.settings' | translate }}</h2>
       <div
         *ngIf="(hasLabelSets | async) === true"
         class="field">
-        <div class="body-m labels">{{ 'upload.labels' | translate }}</div>
-        <app-label-field
-          [labelSets]="labelSets | async"
-          [(selection)]="selectedLabels"></app-label-field>
+        <label
+          class="body-m"
+          for="label-expander">
+          {{ 'upload.labels' | translate }}
+        </label>
+        <div class="labels-container">
+          <nsi-labels-expander
+            id="label-expander"
+            [labelSets]="labelSets | async"
+            [currentSelection]="currentSelection"
+            (updateSelection)="updateLabelSelection($event)"></nsi-labels-expander>
+          <div>
+            <app-label-list
```

Comment:
I created a connector with some labels and works fine.
Later if go back to the connector settings page, the label list is displayed empty

Reply:
@operramon fixed